pkg/terraform/terraform.go

```diff
@@ -25,14 +25,31 @@ func newTFExec(datadir string, terraformDir string) (*tfexec.Terraform, error) {
 		return nil, err
 	}
 
-	lpDebug := &lineprinter.LinePrinter{Print: (&lineprinter.Trimmer{WrappedPrint: logrus.Debug}).Print}
-	lpError := &lineprinter.LinePrinter{Print: (&lineprinter.Trimmer{WrappedPrint: logrus.Error}).Print}
-	defer lpDebug.Close()
-	defer lpError.Close()
+	// FIXME: just setting this here so we don't have to make changes to CI
+	os.Setenv("TF_LOG_PATH", os.ExpandEnv("${ARTIFACT_DIR}/terraform.log"))
+	os.Setenv("TF_LOG_CORE", "info")
+	os.Setenv("TF_LOG_PROVIDER", "debug")
 
-	tf.SetStdout(lpDebug)
-	tf.SetStderr(lpError)
-	tf.SetLogger(newPrintfer())
+	//  Do not set Stderr and Stdout for terraform-exec otherwise the logrus
+	//  hook is going to capture terraform logs and add them to the installer
+	//  log. Terraform logs should have timestamps of their own, so we are just
+	//  adding unnecessary noise
+	if path, ok := os.LookupEnv("TF_LOG_PATH"); ok {
+		tf.SetLogPath(path)
+		tf.SetLog(os.Getenv("TF_LOG"))
+		tf.SetLogCore(os.Getenv("TF_LOG_CORE"))
+		tf.SetLogProvider(os.Getenv("TF_LOG_PROVIDER"))
+	} else {
+		// Add terraform info logs to the installer log
+		lpDebug := &lineprinter.LinePrinter{Print: (&lineprinter.Trimmer{WrappedPrint: logrus.Debug}).Print}
+		lpError := &lineprinter.LinePrinter{Print: (&lineprinter.Trimmer{WrappedPrint: logrus.Error}).Print}
+		defer lpDebug.Close()
+		defer lpError.Close()
+
+		tf.SetStdout(lpDebug)
+		tf.SetStderr(lpError)
+		tf.SetLogger(newPrintfer())
```

Comment:
This is the part I'm not happy about. It seems the output we get from these is not the same we get from the log that goes into the file. I'm wondering if it's really a good idea to leave these out when we're sending tf logs to a file. Maybe we should still keep them in the `openshift_install.log`?

Reply:
best to leave it to maximize the logs.

Reply:
Done